src/os/ui/draw/basedrawcontrols.js

```diff
@@ -114,12 +110,34 @@ os.ui.draw.BaseDrawControlsCtrl = function($scope, $element) {
    * If the line control is supported.
    * @type {boolean}
    */
-  this['supportsLines'] = this['supportsLines'] || false;
+  this['supportsLines'] = false;
+
+  /**
+   * If extra controls should be hidden.
+   * @type {boolean}
+   */
+  this['hideExtraControls'] = false;
 
   /**
    * @type {os.ui.menu.Menu|undefined}
    */
   this['controlMenu'] = os.ui.menu.draw.MENU;
+};
+
+
+/**
+ * The logger.
+ * @const
+ * @type {goog.log.Logger}
+ * @private
+ */
+os.ui.draw.BaseDrawControlsCtrl.LOGGER_ = goog.log.getLogger('os.ui.draw.BaseDrawControlsCtrl');
+
+
+/**
+ * Angular initialization lifecycle function. Sets up event listening and our controls menu.
+ */
+os.ui.draw.BaseDrawControlsCtrl.prototype.$onInit = function() {
```

Comment:
To be clear about the `$onInit` function, not every directive needs to have one. Typically it's useful for stuff that you want to have happen after the directive gets added to the DOM and rendered for the first time. It's useful for timing/sizing issues, and probably not necessary for other situations. It probably won't hurt anything to have it here in this case, but unless something was broken without it, it shouldn't be necessary to add it.

Reply:
I should have left comments in this ticket instead of Slack, but this particular `$onInit` is for the case where we want code to execute after the class has been fully constructed. This allows the child constructor to influence the initialization function without overriding it. In this case, the child is changing the `supportsLines` property which impacts `initControlMenu`.